src/Microsoft.Data.SqlClient/netcore/src/Microsoft/Data/SqlClientX/Handlers/Connection/ConnectionHandlerContext.cs

```diff
@@ -58,12 +59,24 @@ internal class ConnectionHandlerContext : HandlerRequest, ICloneable
         /// Whether the connection is capable of MARS
         /// This is negotiated after pre-login.
         /// </summary>
-        public bool MarsCapable { get; internal set; }
+        public bool IsMarsCapable { get; internal set; }
 
         /// <summary>
         /// Indicates if fed auth needed for this connection.
         /// </summary>
-        public bool FedAuthRequired { get; internal set; }
+        public bool IsFedAuthRequired { get; internal set; }
+
+        /// <summary>
+        /// Get if data classification is enabled by the server.
+        /// </summary>
+        public bool IsDataClassificationEnabled =>
+                (DataClassificationVersion != TdsEnums.DATA_CLASSIFICATION_NOT_ENABLED);
+
+        /// <summary>
+        /// Get or set data classification version.
+        /// A value of 0 means that sensitivity classification is not enabled.
+        /// </summary>
+        public int DataClassificationVersion { get; set; }
```

Comment:
I am hoping that we have a type called `ConnectionFeatures` in the context, and the ConnectionFeatures type has the details of each of the FE. Each of the feature would contain 
1. Whether it was negotiated.
2. Whether the server acknowledged it
3. The version (if any) of the feature acknowledged by the server.
This way the responsibilities of all the features end up in a different object, instead of rooting them at the connection handler context.

Reply:
I have something like this in mind https://github.com/saurabh500/SqlClient/blob/dev/loginhandler2/src/Microsoft.Data.SqlClient/netcore/src/Microsoft/Data/SqlClientX/Handlers/Connection/Login/FeatureExtensions.cs

This will come in with LoginHandler